plugins/airbrake/src/components/EntityAirbrakeWidget/EntityAirbrakeWidget.tsx

```diff
@@ -36,59 +37,70 @@ const useStyles = makeStyles<BackstageTheme>(() => ({
   },
 }));
 
+enum ComponentState {
+  Loading,
+  NoProjectId,
+  Error,
+  Loaded,
+}
+
 export const EntityAirbrakeWidget = ({ entity }: { entity: Entity }) => {
   const classes = useStyles();
 
   const projectId = useProjectId(entity);
-  const errorApi = useApi<ErrorApi>(errorApiRef);
   const airbrakeApi = useApi(airbrakeApiRef);
+  let componentState = ComponentState.Loading;
 
-  const { loading, value, error } = useAsync(
-    () => airbrakeApi.fetchGroups(projectId),
-    [airbrakeApi, projectId],
-  );
-
-  useEffect(() => {
-    if (error) {
-      errorApi.post(error);
-    }
-  }, [error, errorApi]);
-
-  if (loading || !projectId || error) {
-    return (
-      <InfoCard title="Airbrake groups" variant="gridItem">
-        {loading && <Progress />}
+  const { loading, value, error } = useAsync(() => {
```

Comment:
Here inside the function you can just do `if (!projectId) { return null; }` or whatever - i disagree with the previous comment, you can absolutely do this and then handle the outcome in the code below, this is all unnecessary. Unless I am gravely mistaken, this whole pull request could be boiled down to roughly four effective lines of code changed if simplified, and turn out to be much more easy to read. :)